core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -4820,12 +4820,8 @@ public boolean isSubjectToMandatoryReadPermissionCheck(String restOfPath) {
         }
 
         // TODO SlaveComputer.doSlaveAgentJnlp; there should be an annotation to request unprotected access
-        if (restOfPath.matches("/computer/[^/]+/slave-agent[.]jnlp")
-            && "true".equals(Stapler.getCurrentRequest().getParameter("encrypt"))) {
-            return false;
-        }
-
-        return true;
+        return !restOfPath.matches("/computer/[^/]+/slave-agent[.]jnlp")
+                || !"true".equals(Stapler.getCurrentRequest().getParameter("encrypt"));
```

Comment:
Maybe you could create private auxiliary methods with a clear name to ease understanding this? Also what does Stapler return? Maybe !"true" could be changed to "false"?